include/boost/compute/context.hpp

```diff
@@ -175,27 +175,13 @@ class context
     /// devices, the first is returned.
     device get_device() const
     {
-        size_t count = 0;
-        clGetContextInfo(m_context,
-                         CL_CONTEXT_DEVICES,
-                         0,
-                         0,
-                         &count);
-        if(count == 0){
-            return device();
-        }
+        std::vector<device> devices = get_devices();
 
-        cl_device_id id;
-        clGetContextInfo(m_context,
-                         CL_CONTEXT_DEVICES,
-                         sizeof(cl_device_id),
-                         &id,
-                         0);
-        if(id == 0){
+        if(devices.empty()) {
             return device();
         }
 
-        return device(id);
+        return devices.front();
```

Comment:
Interesting, what hardware/platform does this occur on?

Just FYI, both this version and the original work fine on the platforms I have (NVIDIA, AMD, Intel).